parser/match_operations.go

```diff
@@ -437,21 +437,48 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
-			}
+		if err := validateMatchOperationsForOppositeComparison(amountMatch[0], matches); err != nil {
+			return err
+		}
+		if err := validateMatchOperationsForOppositeComparison(amountMatch[1], matches); err != nil {
+			return err
+		}
+		// check for opposites amount
+		if err := oppositeAmounts(
+			matches[amountMatch[0]].Operations[0],
+			matches[amountMatch[1]].Operations[0],
+		); err != nil {
+			return fmt.Errorf("%w: amounts not opposites", err)
 		}
 	}
 
 	return nil
 }
 
+// validateMatchOperationsForOppositeComparison checks if the matched operations have at least one element and
+// checks if all the operations have the same amount
+func validateMatchOperationsForOppositeComparison(matchIndex int, matches []*Match) error {
+	matchOps := matches[matchIndex].Operations
+
+	if len(matchOps) == 0 {
+		return fmt.Errorf(
+			"opposite amounts comparison error: no matched operations for match index %d",
+			matchIndex,
+		)
+	}
+	// check if all operations within the same match slice have the same amount
+	if len(matchOps) > 1 {
```

Comment:
nit: you can remove this if statement as we assume it must be true (assuming `matchIndex` is not negative, which it should never be)

Reply:
equalAmounts method will return err if length of the operations is less than 1.
https://github.com/coinbase/rosetta-sdk-go/blob/master/parser/match_operations.go#L304-L306

Reply:
We already check if `len(matchOps) < 1` on line 467 in your PR.